packages/components/src/components/hds/code-block/index.ts

```diff
@@ -97,22 +135,22 @@ export default class HdsCodeBlockIndexComponent extends Component {
    * @default false
    * @description Make text content wrap on multiple lines
    */
-  get hasLineWrapping() {
+  get hasLineWrapping(): boolean {
     return this.args.hasLineWrapping ?? false;
   }
 
   @action
-  setPrismCode(element) {
+  setPrismCode(element: HTMLElement): void {
     const code = this.code;
     const language = this.language;
-    const grammar = Prism.languages[language];
+    const grammar = language ? Prism.languages[language] : undefined;
 
     if (code) {
       next(() => {
         if (language && grammar) {
           this.prismCode = htmlSafe(Prism.highlight(code, grammar, language));
         } else {
-          this.prismCode = htmlSafe(Prism.util.encode(code));
+          this.prismCode = htmlSafe(Prism.util.encode(code).toString());
```

Comment:
Question: this is the one bit I'm not super sure works as expected. `Prism.util` returns a type TokenString which doesn't match string which `htmlSafe` expects. Looking at the showcase example for no language set (https://hds-showcase-git-hds-3550-ts-codeblock-hashicorp.vercel.app/components/code-block#language), it seems fine but not sure if other code snippets could break.

Reply:
the "hasLineWrapping=true with long string" example also doesn't have a language set and it works: https://hds-showcase-git-hds-3550-ts-codeblock-hashicorp.vercel.app/components/code-block#line-wrapping